plugins/inputs/tacacs/tacacs_test.go

```diff
@@ -0,0 +1,255 @@
+package tacacs
+
+import (
+	"context"
+	"net"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/nwaples/tacplus"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+type testRequestHandler map[string]struct {
+	password string
+	args     []string
+}
+
+func (t testRequestHandler) HandleAuthenStart(_ context.Context, a *tacplus.AuthenStart, s *tacplus.ServerSession) *tacplus.AuthenReply {
+	user := a.User
+	for user == "" {
+		c, err := s.GetUser(context.Background(), "Username:")
+		if err != nil || c.Abort {
+			return nil
+		}
+		user = c.Message
+	}
+	pass := ""
+	for pass == "" {
+		c, err := s.GetPass(context.Background(), "Password:")
+		if err != nil || c.Abort {
+			return nil
+		}
+		pass = c.Message
+	}
+	if u, ok := t[user]; ok && u.password == pass {
+		return &tacplus.AuthenReply{Status: tacplus.AuthenStatusPass}
+	}
+	return &tacplus.AuthenReply{Status: tacplus.AuthenStatusFail}
+}
+
+func (t testRequestHandler) HandleAuthorRequest(_ context.Context, a *tacplus.AuthorRequest, _ *tacplus.ServerSession) *tacplus.AuthorResponse {
+	if u, ok := t[a.User]; ok {
+		return &tacplus.AuthorResponse{Status: tacplus.AuthorStatusPassAdd, Arg: u.args}
+	}
+	return &tacplus.AuthorResponse{Status: tacplus.AuthorStatusFail}
+}
+
+func (t testRequestHandler) HandleAcctRequest(_ context.Context, _ *tacplus.AcctRequest, _ *tacplus.ServerSession) *tacplus.AcctReply {
+	return &tacplus.AcctReply{Status: tacplus.AcctStatusSuccess}
+}
+
+func TestTacacsLocal(t *testing.T) {
+	testHandler := tacplus.ServerConnHandler{
+		Handler: &testRequestHandler{
+			"testusername": {
+				password: "testpassword",
+			},
+		},
+		ConnConfig: tacplus.ConnConfig{
+			Secret: []byte(`testsecret`),
+			Mux:    true,
+		},
+	}
+	l, err := net.Listen("tcp", "localhost:1049") // Use port 1049 instead of 49, so we are above 1023 for unix tests
+	require.NoError(t, err, "local net listen failed to start listening on port 1049")
+
+	srv := &tacplus.Server{
+		ServeConn: func(nc net.Conn) {
+			testHandler.Serve(nc)
+		},
+	}
+
+	go func() {
+		err = srv.Serve(l)
+		require.NoError(t, err, "local srv.Serve failed to start serving on port 1049")
+	}()
+
+	plugin := &Tacacs{
+		Servers:         []string{"localhost:1049"},
+		Username:        config.NewSecret([]byte(`testusername`)),
+		Password:        config.NewSecret([]byte(`testpassword`)),
+		Secret:          config.NewSecret([]byte(`testsecret`)),
+		RequestAddr:     "127.0.0.1",
+		ResponseTimeout: config.Duration(0),
+		Log:             testutil.Logger{},
+	}
+	var acc testutil.Accumulator
+	require.NoError(t, plugin.Init())
```

Comment:
added some Init tests (tried also default timeout but thats set in init() instead of Init() so its not called in tests, would need more effort to implement test for that)

Reply:
I think its best to have that in a separate test like `TestInit`..

Reply:
yea makes it more readable